src/identifyNode.ts

```diff
@@ -1,20 +1,56 @@
 let AWS = require('aws-sdk');
 
 export function handler(event, context, callback) {
-    console.log("Searching for oldest node...");
-    describeAsg("asg123");
-    callback(null, "Success");
+    let asg: string = process.env.ASG_NAME;
+    console.log(`Searching for oldest node in ${asg}`);
+    identifyOldestInstance(asg, callback);
 }
 
-let asg = new AWS.AutoScaling();
+let awsAutoscaling = new AWS.AutoScaling();
+let awsEc2 = new AWS.EC2();
 
-function describeAsg(asgName: string) {
+function identifyOldestInstance(asgName: string, callback) {
     let params = { AutoScalingGroupNames: [ asgName ] };
-    asg.describeAutoScalingGroups(params, function(error, data) {
-       if (error) console.log(error, error.stack, error.statusCode);
-       else console.log(data);
+    awsAutoscaling.describeAutoScalingGroups(params, function(error, data) {
+       if (error) {
+           console.log(error, error.stack, error.statusCode);
+       } else {
+           let instances = data.AutoScalingGroups[0].Instances;
+           let ids = instances.map(instance => instance.InstanceId);
+           getOldestInstanceId(ids, callback);
```

Comment:
rather than passing the callback all the way through, we could return promises and chain them together, something like
```
getInstanceIds(asg)
  .then(ids => getOldestInstanceId(ids))
  .then(id => callback(null, { "oldestInstance": id}))
  .catch(error => callback(error))
```